dig_test.go

```diff
@@ -3741,6 +3741,119 @@ func TestProvideInfoOption(t *testing.T) {
 	})
 }
 
+func TestInvokeInfoOption(t *testing.T) {
```

Comment:
These tests can easily be, and should be, consolidated to a table test. 

Also, some of these tests don't seem like they're necessary. i.e. why do you need the "two invokes" test? that's not testing a different code path that doesn't get exercised by a single invoke, afaict.

Reply:
One easy way  to do this is:
declare a container and add a bunch of provides,
then build a table of `struct{ desc string; invoke any; want *InvokeInfo }`
where `invoke any` holds the function to invoke.

To keep things simple and more readable, you don't need to bother with the `require.NotNil` inside the function bodies. We've already exhaustively tested Dig, so this test can be limited to just testing the InvokeInfo functionality.

```go
c := digtest.New(t)
c.Provide(...)
// .. a few more depending on what you need.
// the provides themselves are not super important
// from the point of view of this test.
// as a reader, all i care about is "given this function signature, this is the invoke info".

tests := []struct {
  desc string
  give any
  want *InvokeInfo
}{
  {
    desc: "one argument",
    give: func(map[string]string) {},
    want: *InvokeInfo{ ... }
  },
  // ...
}

for _, tt := range tests {
  // ...
}
```